compiler/plugins/target/AMD-AIE/aie/AMDAIEObjectFifoStatefulTransform.cpp

```diff
@@ -284,589 +201,460 @@ void createDMA(
+  // side.
+  std::swap(lockPair.first, lockPair.second);
+  std::pair<uint8_t, uint8_t> lockAcqRel = std::make_pair(1, 1);
+  resourceMap[createOp].producerResource = ObjectFifoEndpointResource(
+      producerBuffers, LockResources(lockPair, lockAcqRel));
+  return success();
+}
 
-  // if split, the necessary size for producer fifo might change
-  SharedMemoryDirection shareDirection = NONE;
-  if (requiresDMAs(createOp, shareDirection, splitBecauseLink)) {
-    IntegerAttr elemNumberAttr;
-    if (isa<ArrayAttr>(createOp.getElemNumber()))
-      elemNumberAttr = builder.getI32IntegerAttr(objFifoSize(createOp));
-    else
-      elemNumberAttr = builder.getI32IntegerAttr(
-          findObjectFifoSize(device, producerTileOp, createOp));
-    createOp.setElemNumberAttr(elemNumberAttr);
+/// Utility to create buffers and locks for the objectFifo consumer side.
+LogicalResult createConsumerBuffersAndLocks(
+    OpBuilder &builder, const AMDAIEDeviceModel &deviceModel,
+    ObjectFifoCreateOp createOp, size_t external_idx,
+    DenseMap<ObjectFifoCreateOp, ObjectFifoResources> &resourceMap) {
+  OpBuilder::InsertionGuard g(builder);
+  resourceMap[createOp].consumerResources.clear();
+  size_t depth = objFifoSize(createOp);
+  for (auto &&[idx1, consumerTile] :
+       llvm::enumerate(createOp.getConsumerTiles())) {
+    size_t idx = external_idx * createOp.getConsumerTiles().size() + idx1;
+    TileOp consumerTileOp =
+        dyn_cast_if_present<TileOp>(consumerTile.getDefiningOp());
+    if (!consumerTileOp) {
+      return createOp.emitOpError()
+             << "expected a consumer tile op, but got: " << consumerTile;
+    }
+    SmallVector<BufferOp> consumerBuffers =
+        createBuffers(builder, deviceModel, createOp, depth, consumerTileOp,
+                      name(createOp).str() + "_cons", idx);
+    std::pair<LockOp, LockOp> lockPair =
+        createLockPair(builder, deviceModel, consumerTileOp, depth,
+                       name(createOp).str() + "_cons", idx);
+    std::pair<uint8_t, uint8_t> lockAcqRel = std::make_pair(1, 1);
+    resourceMap[createOp].consumerResources[consumerTileOp] =
+        ObjectFifoEndpointResource(consumerBuffers,
+                                   LockResources(lockPair, lockAcqRel));
   }
+  return success();
+}
 
-  if (!objFifoSize(createOp)) return;
+LogicalResult createBuffersAndLocks(
+    OpBuilder &builder, DeviceOp device, ObjectFifoLinkOp linkOp,
+    DenseMap<ObjectFifoCreateOp, ObjectFifoResources> &resourceMap) {
+  OpBuilder::InsertionGuard g(builder);
+  AMDAIEDeviceModel deviceModel =
+      getDeviceModel(static_cast<AMDAIEDevice>(device.getDevice()));
 
-  auto fifo = llvm::cast<AIEObjectFifoType>(createOp.getElemType());
-  auto elemType = llvm::cast<MemRefType>(fifo.getElementType());
+  std::vector<ObjectFifoCreateOp> inputs = getInputObjectFifos(linkOp);
+  std::vector<ObjectFifoCreateOp> outputs = getOutputObjectFifos(linkOp);
```

Comment:
Any strong reason(s) for preferring `std::vector` to `SmallVector` ? I believe the latter is preferred generally.

Reply:
No, this is a leftover from copy pasting code. Thanks for pointing out, I updated all of them.